### PR TITLE
Feature: user permissions

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/main/resources/db/migration/V1_3_0__new_table_access_controls_tree.sql [hunk 1]
```diff
@@ -7,6 +7,7 @@ CREATE TABLE IF NOT EXISTS access_controls_tree (
     UNIQUE (parent, current)
 );
 
+
```
Comment (justinehell) [inline src/main/resources/db/migration/V1_3_0__new_table_access_controls_tree.sql:10]: why add blank line?

File: src/main/java/com/ditrit/letomodelizerapi/persistence/model/UserPermission.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * The identifier of the user to whom this permission is assigned.
+     */
+    @Column(name = "usr_id")
+    @FilterType(type = FilterType.Type.NUMBER)
+    private Long userId;
+
+    /**
+     *  The entity associated with this permission.
+     */
```
Comment (justinehell) [inline src/main/java/com/ditrit/letomodelizerapi/persistence/model/UserPermission.java:40]:
```suggestion
    /**
     * The entity associated with this permission.
     */
```

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V1_3_0__new_table_access_controls_tree.sql (hunk 1 above)
Comment (Zorin95670) [reply src/main/resources/db/migration/V1_3_0__new_table_access_controls_tree.sql]: bad rebasing, done.

File: src/main/java/com/ditrit/letomodelizerapi/persistence/model/UserPermission.java (hunk 2 above)
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/persistence/model/UserPermission.java]: done

**Review by justinehell — APPROVED** [review]

File: src/main/java/com/ditrit/letomodelizerapi/persistence/repository/UserPermissionRepository.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * Retrieves a list of {@link UserPermission} entities for a specific user
+     * where the entity type does not match the specified filter.
+     *
+     * @param userId The ID of the user whose permissions are being queried.
+     * @param entityFilter The entity type to be excluded from the results.
+     * @return A list of {@link UserPermission} entities matching the criteria.
+     */
+    List<UserPermission> findAllByUserIdAndEntityIsNot(Long userId, String entityFilter);
```
Comment (aboubakar14) [inline src/main/java/com/ditrit/letomodelizerapi/persistence/repository/UserPermissionRepository.java:26]: I don't like the name of the methods, very ugly

**Review by aboubakar14 — APPROVED** [review]

**Pull request merged**